### ISSUE TITLE
IndexSelecter scaling issue when using DateScale

### ISSUE THREAD

**Issue by candronikos** [bqplot/bqplot#767]
There is an issue with the `IndexSelector` when used with `DateScale`. The datetime seen in the callback doesn't match the point in time where the mouse is hovering.
In the `IndexSelector` example I get The selected date is `['2006-11-14T02:13:42.656000']` when the mouse is hovering over the first of January.

```
import pandas as pd
import numpy as np

symbol = 'Security 1'
symbol2 = 'Security 2'

price_data = pd.DataFrame(np.cumsum(np.random.randn(150, 2).dot([[0.5, 0.4], [0.4, 1.0]]), axis=0) + 100,
                          columns=['Security 1', 'Security 2'],
                          index=pd.date_range(start='01-01-2007', periods=150))

dates_actual = price_data.index.values
prices = price_data[symbol].values

from bqplot import DateScale, LinearScale, Axis, Lines, Scatter, Bars, Hist, Figure
from bqplot.interacts import (
    FastIntervalSelector, IndexSelector, BrushIntervalSelector,
    BrushSelector, MultiSelector, LassoSelector, PanZoom, HandDraw
)
from traitlets import link

from ipywidgets import ToggleButtons, VBox, HTML

## First we define a Figure
dt_x_fast = DateScale()
lin_y = LinearScale()

x_ax = Axis(label='Index', scale=dt_x_fast)
x_ay = Axis(label=(symbol + ' Price'), scale=lin_y, orientation='vertical')
lc = Lines(x=dates_actual, y=prices, scales={'x': dt_x_fast, 'y': lin_y}, colors=['orange'])
lc_2 = Lines(x=dates_actual[50:], y=prices[50:] + 2, scales={'x': dt_x_fast, 'y': lin_y}, colors=['blue'])

db_index = HTML(value='[]')

## Now we try a selector made to select all the y-values associated with a single x-value
index_sel = IndexSelector(scale=dt_x_fast, marks=[lc, lc_2])

## Now, we define a function that will be called when the selectors are interacted with
def index_change_callback(change):
    db_index.value = 'The selected date is ' + str(change.new)

index_sel.observe(index_change_callback, names=['selected'])

fig_index_sel = Figure(marks=[lc, lc_2], axes=[x_ax, x_ay], title='Index Selector Example',
                       interaction=index_sel)
display(VBox([db_index, fig_index_sel]))
display(x_ax,x_ay)
```

**Comment by jasongrout** [comment]
CC @rmenegaux

**Comment by DougRzz** [comment]
A few ideas that may or may not fix the issue.  

1) 
```python
dt_x_fast.allow_padding = False`
``` 
This ensures the axis starts at precisely 1st Jan with no padding.  You may have been hovering over a date before the 1st Jan.

2)  The time zone that your computer is set to, if not UTC,   might mess up the date output by a few hours.  See another issue... 
https://github.com/bloomberg/bqplot/issues/617

3) There are a finite number of pixels on the screen with may affect the precision.

**Comment by rmenegaux** [comment]
Hi @candronikos, which version of bqplot are you using, and in what browser? The example works fine on my side

**Comment by joders** [comment]
This issue is likely fixed by either 
fix: no timezone jump on PanZoom actions #969 
or
fix: DateScale doesn't apply browser timezone for ticks anymore #968

**Comment by SylvainCorlay** [comment]
Closing as fixed by @joders in #968.
